lib/util/Components.js

```diff
@@ -41,15 +41,18 @@ function usedPropTypesAreEquivalent(propA, propB) {
 }
 
 function mergeUsedPropTypes(propsList, newPropsList) {
-  const propsToAdd = [];
-  newPropsList.forEach((newProp) => {
-    const newPropisAlreadyInTheList = propsList.some((prop) => usedPropTypesAreEquivalent(prop, newProp));
-    if (!newPropisAlreadyInTheList) {
-      propsToAdd.push(newProp);
+  if (propsList === undefined || propsList.length === 0) {
+    return newPropsList;
+  }
```

Comment:
when would `propsList` ever be undefined? the previous code assumed it was always an array.

Reply:
All the callsites for `mergedUsedPropTypes` have been reworked to avoid passing in empty arrays, e.g.
```
        usedPropTypes: mergeUsedPropTypes(component.usedPropTypes || [], props.usedPropTypes || [])
```
into
```
        usedPropTypes: mergeUsedPropTypes(component.usedPropTypes, props.usedPropTypes) || []
```

This means less array copies, and we can do the least amount of work - if you're calling `mergeUsedPropTypes(['a', 'b', 'c'], undefined)` it'll directly return the first arg, instead of creating an extra copy.